docs/pages/api-reference/liveblocks-react.mdx

```diff
@@ -1284,10 +1288,60 @@ const status = useStatus();
 ```ts
 import { useStorageStatus } from "@liveblocks/react/suspense";
 
-const status = useStatus();
+const storageStatus = useStorageStatus();
 // "synchronizing" | "synchronized"
 ```
 
+#### Prevent users losing unsaved changes [#prevent-users-losing-unsaved-changes]
+
+Storage usually synchronizes milliseconds after a change, but in some
+circumstances, such as a user with a poor connection, it may take longer before
+they receive a response from our servers. It can be helpful to check if Storage
+changes have been synchronized, and prevent users from leaving the page if they
+haven’t.
+
+```tsx
+import { useStorageStatus } from "@liveblocks/react/suspense";
+
+function usePreventUnsavedChanges() {
+  const storageStatus = useStorageStatus();
+
+  useEffect(() => {
+    function beforeUnload(e: BeforeUnloadEvent) {
+      if (storageStatus === "synchronized") {
+        return;
+      }
+      e.preventDefault();
+    }
+
+    window.addEventListener("beforeunload", beforeUnload);
+
+    return () => {
+      window.removeEventListener("beforeunload", beforeUnload);
+    };
+  }, [storageStatus]);
```

Comment:
Not sure if I would prefer this implementation. I'd recommend the following implementation:

```suggestion
  const room = useRoom();

  useEffect(() => {
    function beforeUnload(e: BeforeUnloadEvent) {
      if (room.getStorageStatus() === "synchronized") {
        return;
      }
      e.preventDefault();
    }

    window.addEventListener("beforeunload", beforeUnload);

    return () => {
      window.removeEventListener("beforeunload", beforeUnload);
    };
  }, [room]);
```

This is because it would save (a lot of) re-renders for a component like this. It may actually be good to call out in the docs that if you use `useStorageStatus()` you should expect lots and lots of rerenders for any component that uses it. This is less true for the `{ smooth: true }` version, of course.